libs/store/src/lib/store/utils/dispatch-data-to-store/dispatch-data-to-store.util.ts

```diff
@@ -1,37 +1,159 @@
 import { Store } from '@ngrx/store';
-import { Observable, throwError } from 'rxjs';
-import { catchError, finalize, tap } from 'rxjs/operators';
+import { Observable, of, throwError } from 'rxjs';
+import { catchError, finalize, switchMap, tap } from 'rxjs/operators';
+import {
+	BaseStoreActions,
+	BaseStoreSelectors,
+	EntityStoreActions,
+	EntityStoreSelectors,
+} from '../../interfaces';
+
+// -------------------
+// Overloads
+// -------------------
+// BASE STORE
+export function dispatchDataToStore<DataType = any>(
+	dispatchAction: BaseStoreActions<DataType>,
+	selectAction: BaseStoreSelectors<DataType>,
+	data: Observable<DataType>,
+	store: Store,
+	dispatchType: 'set'
+): Observable<DataType>;
+export function dispatchDataToStore<DataType = any>(
+	dispatchAction: BaseStoreActions<DataType>,
+	selectAction: BaseStoreSelectors<DataType>,
+	data: Observable<DataType>,
+	store: Store
+): Observable<DataType>;
+// NO SELECTOR
+export function dispatchDataToStore<DataType = any>(
+	dispatchAction: BaseStoreActions<DataType>,
+	selectAction: null | undefined,
+	data: Observable<DataType>,
+	store: Store,
+	dispatchType: 'set'
+): Observable<null>;
+export function dispatchDataToStore<DataType = any>(
+	dispatchAction: BaseStoreActions<DataType>,
+	selectAction: null | undefined,
```

Comment:
I'm not that familiar with these overloads, but what will be the effect of putting the selectAction here on the existing implementation?

It's one thing that the result of this PR forces you to change the return types, but completely changing the order of this implementation forcing you to pass undefined as a second parameter would make it a lot harder to work wit.

I'd personally expect the overload to be either with the dispatchAction and one without it?

Reply:
Yes, that would be possible, I presume! 👍